.github/actions/setup-ubuntu/action.yml

```diff
@@ -0,0 +1,22 @@
+name: 'setup Ubuntu'
+description: 'Setup an Ubuntu container for running CI'
+runs:
+  using: 'composite'
+  steps:
+    - name: Install dependencies
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install --yes \
```

Comment:
> _Fwiw_ one can use "apt-get build-dep kmod" as a shorter alternative.

That works as  long the distro is up to date on deps with latest in-development things. It also requires to add the src repos afair. I think we can eventually move to build-dep, but I like the finer control for now of the deps we install.

> Out of curiosity: are there any benefits of using nested/composite yml files, instead of a shell script?

I wanted to keep this step separate from anything else. I think I could have used a ./github/setup-ubuntu.sh instead of an action without problems. Only benefit of having an action is that it's automatically usable by other projects (example: if we want an additional repo with validation stuff in kmod-project/*, we could simply use the same action in its github CI.  Not a huge benefit and I didn't really did like that because of this.